File: android/sdk/src/main/java/com/wya/hybrid/localserver/LocalService.java
package com.wya.hybrid.localserver;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;
import android.util.Log;

import java.io.IOException;
/**
 * @author :
 */
public class LocalService extends Service {
    private LocalServer mLocalServer;
    
    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
    
    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        mLocalServer = new LocalServer(getApplication());
        try {
            mLocalServer.start();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return super.onStartCommand(intent, flags, startId);
    }
    
    @Override
    public void onDestroy() {
        super.onDestroy();
        if (mLocalServer != null) {
            mLocalServer.closeAllConnections();
            mLocalServer.stop();
        }
    }
    
}

File: android/sdk/src/main/java/com/wya/hybrid/localserver/LocalService.kt
package com.wya.hybrid.localserver

import android.app.Service
import android.content.Intent
import android.os.IBinder
import java.io.IOException

/**
 * @author :
 */
class LocalService : Service() {

	private lateinit  var mLocalServer: LocalServer

	override fun onBind(intent: Intent): IBinder? {
        return null
    }

	override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        mLocalServer = LocalServer(application)
        try {
            mLocalServer.start()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        super.onDestroy()
            mLocalServer.closeAllConnections()
            mLocalServer.stop()
    }

}
